### DISCUSSION TITLE
cannot POST to nextjs api route

### DISCUSSION THREAD

**Discussion by hainessss** [vercel/next.js#19856] [Help] [answered]
completely stumped on this.  

given a brand new project. `yarn create next-app`.

and this basic script. 

```
var request = require('request');

(async function() {
  var options = {
    'method': 'POST',
    'url': 'http://localhost:3000/api/hello',
    'headers': {
      'Content-Type': 'application/json',
    }
  };
  request(options, function (error, response, body) {
    if (error) throw new Error(error);
    console.log(response.toJSON());
  });
})();
```

I get the response:
```
<!DOCTYPE html>\n<html lang="en">\n<head>\n<meta charset="utf-8">\n<title>Error</title>\n</head>\n<body>\n<pre>Cannot POST /api/hello</pre>\n</body>\n</html>\n
```
What am I missing here? It works fine in postman and curl.  

Thanks in advance!

**Comment by mnlfischer** [comment, marked as answer]
Please provide more informations.
Where did you add that script? Where you want to make that request happen?
How your api is implemented? Code snippet would be great.

**Reply by hainessss** [reply]
just added it at the route of the project. can run it by running `node fetchScript.js`.  I am seeing this error in the getserversideprops of my project as well.

**Reply by hainessss** [reply]
I had another server on the same port.  I hate my life. 😆😵

**Reply by leerob** [reply]
@hainessss It happens, don't sweat it 😄

**Reply by mnlfischer** [reply]
good catch! 👍 😄

**Comment by salad-henchmen** [comment]
Also, I notice that the call is local. You don't have to fetch data through api calls from get getserversideprops if you have the data on your server.
